lib/src/client.dart

```diff
@@ -521,13 +536,40 @@ class Client extends MatrixApi {
             loginTypes.map((f) => f.type ?? '').toSet(), supportedLoginTypes);
       }
 
-      return (wellKnown, versions, loginTypes);
+      return (versions, loginTypes);
     } catch (_) {
       homeserver = null;
       rethrow;
     }
   }
 
+  /// Gets discovery information about the domain. The file may include
+  /// additional keys, which MUST follow the Java package naming convention,
+  /// e.g. `com.example.myapp.property`. This ensures property names are
+  /// suitably namespaced for each application and reduces the risk of
+  /// clashes.
+  ///
+  /// Note that this endpoint is not necessarily handled by the homeserver,
+  /// but by another webserver, to be used for discovering the homeserver URL.
+  ///
+  /// In case is provided, [hostname] will be used in place of [homeserver] for
+  /// the HTTPS lookup. This is useful since the matrix homeserver does not
+  /// necessarily also host a copy of the `.well-known`.
+  ///
+  /// The result of this call is stored in [wellKnown] for later use at runtime.
+  @override
+  Future<DiscoveryInformation> getWellknown([String? hostname]) async {
+    final wellKnown = await getDiscoveryInformationsByUserId(
+      hostname ?? baseUri!.host,
+    );
+
+    // do not reset the well known here, so super call
+    super.homeserver = wellKnown.mHomeserver.baseUrl.stripTrailingSlash();
```

Comment:
Changing the homeserver here might be a bit dangerous?  We already seem to set the homeserver in the checkHomeserver function, so for one this seems redundant, but if some developer actually passes a custom hostname to this function, it might break the client completely in an unexpected manner.

In general I would give this function only one purpose, i.e. "refreshWellknown" instead of "getWellknown". Then you could extract the hostname from the userid and request that and update the database. I don't think SDK users should be able to publicly call this with a different hostname.